collectors/scm/github-graphql/src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java

```diff
@@ -1042,6 +1056,42 @@ private static DateTime getDate(DateTime dateInstance, int offsetDays, int offse
         return dateInstance.minusDays(offsetDays).minusMinutes(offsetMinutes);
     }
 
+
+    private JSONObject getDataFromRestCallPost(String url, GitHubRepo repo, String password, String personalAccessToken, JSONObject query) throws MalformedURLException, HygieiaException {
+        ResponseEntity<String> response = makeRestCallPost(url, repo.getUserId(), password, personalAccessToken, query);
+        JSONObject data = (JSONObject) parseAsObject(response).get("data");
+        JSONArray errors = getArray(parseAsObject(response), "errors");
+
+        if (!CollectionUtils.isEmpty(errors)) {
+            JSONObject error = (JSONObject) errors.get(0);
+            if (error.get("type").equals("NOT_FOUND")) {
+                RedirectedStatus redirectedStatus = checkForRedirectedRepo(repo);
+
+                if (redirectedStatus.isRedirected()) {
+                    String redirectedUrl = redirectedStatus.getRedirectedUrl();
+                    LOG.debug("Repo was redirected from: " + repo.getRepoUrl() + " to " + redirectedUrl);
+                    repo.setRepoUrl(redirectedUrl);
```

Comment:
> I think, we should update the corresponding collector item with the "new" home of the repo so that going forward, the first try will be successful and also all other associated data in Hygieia will be clean.

Not sure how the collector items are associated to the repos and I will review this with Nireesh, but I update the repo url here to point to the redirected repo and each run after that only looks at the "new" home of the repo.

Reply:
Redirection sometimes is used as a normal practice, to hide URLs that are often changes, so not sure if it is a good idea to change to the redirected URL permanently. We may want to simply log the redirections, and manually confirm whether the change is permanent or not.

Reply:
@tabladrum @andrewalvintran The collectorItem appears to be updated. However, the component is not be updated to reflect that change.

Reply:
@benj58xu I can't see an instance where a team would change orgs or repo names often.

Reply:
@Sbrenthughes As to redirection, I do not know much about the practices of our interfacing systems, but there are multiple usages in general. For instance, if there are multiple micro services serving client requests directly, we can provide a single URL to the clients and let a proxy to dispatch the requests to the appropriate micro services. In this way, when we add/change micro services, the clients are not impacted.

Reply:
I don't see how this applies to users changing their repo/ org name.

Reply:
If we only need to deal with that particular use case, I agree. According to HTTP spec, they should use 301 in this case, never 302.